doc/nrf/libraries/shell/shell_nfc.rst

```diff
@@ -0,0 +1,132 @@
+The NFC shell answer with the payload started with the same byte as received from the Reader/Writer.
+
+NFC data exchange
+=================
+
+The NFC Reader/Writer device needs to poll for a data constantly.
+There might be a cases when both devices do not have any data to exchange or one of them does not have any data.
+The NFC Tag always respond to the Reader/Writer frame.
+A payload always start with one dummy byte header.
+Its value is not relevant and it is not checked by this library.
+If one of a device does not have any data then only header byte is sent.
+
+The following diagram illustrates different data exchange cases:
+
+.. msc::
+    hscale = "1.3";
+    a [label="Reader/Writer"], b [label="Tag"];
+    |||;
+    a rbox b [label = "Empty packet exchange(poll for data)"];
+    a=>b [label = "data: ['S'] - one dummy byte, 'S' character here"];
+    a<=b [label = "data: ['S'] - one dummy byte"];
+    |||;
+    a rbox b  [label = "Only Reader\Writer has data"];
+    a=>b [label = "data: ['S', 1..N bytes]"];
+    a<=b [label = "data: ['S'] - one dummy byte"];
+    |||;
+    a rbox b [label = "Only Reader\Writer has data"];
+    a=>b     [label = "data: ['S'] - one dummy byte"];
+    a<=b     [label = "data: ['S', 1..N bytes]"];
+    |||;
+    a rbox b [label = "Both devices have data"];
+    a=>b [label = "data: ['S', 1..N bytes]"];
+    a<=b [label = "data: ['S', 1..N bytes]"];
+
+Requirements
+************
+
+This library has a following requirements:
+
+  * The :kconfig:option:`CONFIG_NFC_THREAD_CALLBACK` must be enabled.
+    The NFC callback function must be called from a thread context.
+    If default configuration it is called from the system workqueue context.
+    This library uses the operating system primitives in the callback function which cannot be
+    used in an interrupt context.
+  * If library logs and the logger are enabled, then the logger should not be used in the synchronous mode.
+    Synchronous logs processing can introduce to big latency in NFC events processing which causing the NFC shell backend being unstable.
+    If you need logs from this library, configure logger to be in the deferred mode by setting the :kconfig:option:`CONFIG_LOG_MODE_DEFERRED` Kconfig option.
+
+Configuration
+*************
+
+Set the :kconfig:option:`CONFIG_SHELL_NFC` Kconfig option to enable the shell backend.
+
+The following configuration options are available for this module:
+
+   * :kconfig:option:`CONFIG_SHELL_NFC_INIT_PRIORITY` sets initialization priority for the NFC shell backend.
+   * :kconfig:option:`CONFIG_SHELL_NFC_BACKEND_TX_BUFFER` sets the NFC T4T APDU response buffer size.
+     It size might depends on your Reader/Writer maxiumum supported APDU size.
+   * :kconfig:option:`CONFIG_SHELL_NFC_BACKEND_TX_RING_BUFFER_SIZE` sets the shell Tx ring buffer size.
+     It is used to buffer data coming from shell interface for the NFC T4T transport, that can send data only on the data reception from the Reader/Writer.
```

Comment:
```suggestion
     It is used to buffer data from the shell interface through the NFC T4T transport, that can send data only on the data reception from the polling device.
```
Could you please clarify the second part of the sentence? 
1. **What** can send data?
2. What is **sending data on the data reception**?

Reply:
!. Transport, data is buffered because it cannot be send right after shell request it. Library need to wait for data reception from polling device and sends s